src/main/java/org/elasticsearch/search/suggest/SuggestParseElement.java

```diff
@@ -44,12 +47,15 @@ public SuggestParseElement(Suggesters suggesters) {
 
     @Override
     public void parse(XContentParser parser, SearchContext context) throws Exception {
-        SuggestionSearchContext suggestionSearchContext = parseInternal(parser, context.mapperService(), context.shardTarget().index(), context.shardTarget().shardId());
+        SuggestionSearchContext suggestionSearchContext = parseInternal(parser, context.mapperService(), context.shardTarget().index(), context.shardTarget().shardId(), context, context);
         context.suggest(suggestionSearchContext);
```

Comment:
wondering.. can we move the copy headers/ctx tot he `SearchContex#suggest` method?